core/types/src/attestation_key.rs

```diff
@@ -0,0 +1,315 @@
+            // quotes it will be 0 otherwise, which also happens to map to
+            // SGX_UNLINKABLE_SIGNATURE.
+            sgx_quote_sign_type_t::SGX_UNLINKABLE_SIGNATURE => Ok(QuoteSignature::UnLinkable),
+            sgx_quote_sign_type_t::SGX_LINKABLE_SIGNATURE => Ok(QuoteSignature::Linkable),
+            v => Err(FfiError::UnknownEnumValue(v.0.into())),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct QuoteLibAttestationKeyId(sgx_ql_att_key_id_t);
+
+impl QuoteLibAttestationKeyId {
+    /// The ID
+    pub fn id(&self) -> Id {
+        self.0.id.into()
+    }
+
+    /// The key version
+    pub fn version(&self) -> Version {
+        self.0.version.into()
+    }
+
+    /// The hash of the MRSIGNER key.
+    ///
+    /// # Errors
+    /// When the length of the hash is not a valid length for a hash,
+    /// [`sgx_ql_att_key_id_t::mrsigner_length`].
+    pub fn mr_signer_key_hash(&self) -> Result<MrSignerKeyHash, FfiError> {
+        match self.0.mrsigner_length {
+            32 => {
+                let mut array: [u8; 32] = [0u8; 32];
+                array.copy_from_slice(&self.0.mrsigner[..32]);
+                Ok(MrSignerKeyHash::Sha256(array))
+            }
+            48 => {
+                let mut array: [u8; 48] = [0u8; 48];
+                array.copy_from_slice(&self.0.mrsigner[..48]);
+                Ok(MrSignerKeyHash::Sha384(array))
+            }
+            _ => Err(FfiError::InvalidInputLength),
+        }
+    }
+
+    /// The product ID
+    pub fn product_id(&self) -> ExtendedProductId {
+        self.0.extended_prod_id.into()
+    }
+
+    /// The config ID
+    pub fn config_id(&self) -> ConfigId {
+        self.0.config_id.into()
+    }
+
+    /// The family ID
+    pub fn family_id(&self) -> FamilyId {
+        self.0.family_id.into()
+    }
+
+    /// The algorithm ID
+    pub fn algorithm_id(&self) -> Result<Algorithm, FfiError> {
+        self.0.algorithm_id.try_into()
+    }
+}
+
+new_type_accessors_impls! {
+    QuoteLibAttestationKeyId, sgx_ql_att_key_id_t;
+}
+
+impl Default for QuoteLibAttestationKeyId {
+    fn default() -> Self {
+        Self(sgx_ql_att_key_id_t {
+            id: 0,
+            version: 0,
+            mrsigner_length: 32,
+            mrsigner: [0u8; 48],
+            prod_id: 0,
+            extended_prod_id: [0u8; 16],
+            config_id: [0u8; 64],
+            family_id: [0u8; 16],
+            algorithm_id: 0,
+        })
+    }
+}
+#[derive(Default, Debug, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct ServiceProviderId([u8; 16]);
+
+new_type_accessors_impls! {
+    ServiceProviderId, [u8; 16];
+}
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct ExtendedAttestationKeyId(sgx_att_key_id_ext_t);
+
+impl ExtendedAttestationKeyId {
+    // The base attestation key
+    pub fn base_attestation_key(&self) -> QuoteLibAttestationKeyId {
```

Comment:
Suggestion shortening to `base_key_id()` or just `base()`.

Reply:
renamed to `base_key_id`